### DISCUSSION TITLE
Are zigbee-herdsman-converters device ts files converted to js files during npm build

### DISCUSSION THREAD

**Discussion by RobLeighton67** [Koenkk/zigbee2mqtt#19616] [General]
Apologies if it's a stupid question but I am currently using a Bookworm Linux build with z2m 1.33.2 (node.js v20, npm v10, typescript v5. 2.2) and after building the code my devices folder has '. js' files instead of '. ts' files?

**Comment by Koenkk** [comment]
That's expected, node.js can only read `.js` files, not `.ts`, so the `.ts` files are transpiled to `.js` during the build.

**Reply by RobLeighton67** [reply]
Thanks @Koenkk for the explanation - I am trying to understand what has changed for the Heiman Emergency Button between 1.28.1 (button press generates mqtt message containing action but no battery state) and 1.33.2 (button press generates mqtt message with battery 0 and no action at all!)
Seems like between the two releases the devices source was migrated in github.com to typescript.

**Comment by RobLeighton67** [comment]
Also, I used to make a local change to add a fingerprint for my white label Heiman button ( Molo Tuya ts0211) and npm run build - is that still okay to do, using the js files?
